pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +383,96 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+# Det Diag Rewrites
+@register_canonicalize
+@register_stabilize
+@node_rewriter([det])
+def rewrite_det_diag_from_eye_mul(fgraph, node):
```

Comment:
I wonder if we should look for mul with contant-folded eye matrices (need not be in this PR). Is that more common than an eye with symbolic inputs that cannot be constant folded? I suspect so.

Or perhaps we should not allow the Eye Op to constant fold until later in the rewrites. This is actually something I feel may be useful in many cases. There are classes of Ops we almost never want to actually evaluate and can be worse to constant fold, or too early at least.

Reply:
What would a mul with a constant-folded eye look like on the graph? It would be folded into a composite?

Reply:
`eye` would be a constant same as starting the graph with `np.eye(5) * pt.vector("x")`, so you would need to check if that first input is a numerical (not symbolic) identity matrix. This is relevant for stuff like arange as well the graph could have `pt.arange` in it or a constant-folded/user provided `np.arange`. It may make sense at some point to unconstant_fold at some stage of the rewrites to get `pt.eye`, `pt.arange` out of constants.